tools/format.py

```diff
@@ -58,11 +58,15 @@ def prettier():
                           "bin-prettier.js")
     source_files = get_sources(root_path, ["*.js", "*.json", "*.ts", "*.md"])
     if source_files:
-        print_command("prettier", source_files)
-        run(["node", script, "--write", "--loglevel=error", "--"] +
-            source_files,
-            shell=False,
-            quiet=True)
+        max_command_length = 32000
+        while len(source_files) > 0:
+            command = ["node", script, "--write", "--loglevel=error", "--"]
+            while len(source_files) > 0:
+                command.append(source_files.pop())
+                if len(" ".join(command)) > max_command_length:
+                    source_files.append(command.pop())
```

Comment:
Thank you @MarkTiedemann! 

This looks pretty good as it is, I have one comment:

I think you're overestimating the accuracy of the length computation a bit.
Keep in mind that these command line arguments might get wrapped in quotes, escaped, converted to a different character set etc, which can all make the final command line longer.

So what I would suggest is to not lower the max_command_length even a bit further (maybe to 25k or so), and don't bother moving the last file that pushed it over the edge back from `command` to `source_files`.

(I'm also saying this because moving a file to another array and back is kinda confusing at first glance, and this utility script is not a great place for complex or clever code.)

Reply:
Thanks for the feedback, @piscisaureus! You are right, I overestimated the accuracy. I will adjust the code accordingly.